src/core/pyproxy.js

```diff
@@ -290,40 +321,93 @@ Module.PyProxyClass = class {
     }
     return Module.hiwire.pop_value(idresult);
   }
-  apply(jsthis, jsargs) {
-    return Module.callPyObject(_getPtr(this), ...jsargs);
+  /**
+   * Test whether the :any:`PyProxy.length` getter is available on this PyProxy. A
+   * Typescript type guard.
```

Comment:
```suggestion
   * Check whether the :any:`PyProxy.length` getter is available on this PyProxy. A
   * Typescript type guard.
```

I understand it's used as a type guard, but this also changes [the API PyProxy by adding these methods](https://pyodide--1601.org.readthedocs.build/en/1601/usage/api/js-api.html#pyproxy), doesn't it? Is this something we want and are willing to maintain in the future?  If so we should add an extra changelog entry.

Same for all such methods.

Reply:
I think these methods are a nice improvement even without Typescript. The nature of the javascript `Proxy` object makes duck typing hard: the alternative would be to check with 
```javascript
x.has && !pyodide.isPyProxy(x.has)
```
which currently leaks memory, is not obvious, and is much slower than the `hasHas` (not that speed is important for this).

Reply:
We could have better names for `hasX`. MyPy seems to use `supportsHas`, do you think that is better?

Reply:
Not just MyPy but also [typing apparently](https://docs.python.org/3/library/typing.html#typing.SupportsBytes). No strong opinions but maybe +1 for `supportsX`